### PR TITLE
WIP: Allow importing and calling main

### PR DESCRIPTION
**Pull request by adelavega**
Here's on option to address #6 

First, I removed the defaults from argparse and put them in `create_workflow`. This actually ends up not mattering much but IMO its cleaner and more explicit

More importantly, `main` now takes an `args` argument which is passed to `parse_args`. 
So now you can do the followng:

    from fitlins.cli.run import main
    main('dataset', 'output_dir', 'dataset', '--model=model.json')

and argparse will parse it. If no argument is passed, command line arguments (from sys) are passed in instead.

There's probably a better way to have a library CLI but for now this is good enough for me.

### CODE REVIEW

**Comment by adelavega** [comment]
Oh also, is this a Python 3 module or Python 2/3? I guess given that its a BIDS app you don't have to care too much about Python 2 compatibility.

File: fitlins/cli/run.py [hunk 1]
```diff
@@ -60,94 +60,102 @@ def get_parser():
...
+
     warnings.showwarning = _warn_redirect
-    opts = get_parser().parse_args()
-    if opts.debug:
-        logger.setLevel(logging.DEBUG)
+    opts = get_parser().parse_args(args)
+    opts = {k:v for k,v in vars(opts).items() if v is not None} # Remove None
 
-    create_workflow(opts)
+    create_workflow(**opts)
```
Comment (effigies) [inline fitlins/cli/run.py]: I think I'd rather replace the whole PR with:

```Diff
- def main():
+ def main(args=None):
...
-     opts = get_parser.parse_args()
+     opts = get_parser.parse_args(args)
```

I think this does what you actually want, without doing a lot of refactoring now. (`parse_args` default value for `args` is already `None`, so there's no real need for checking values and branching.)

While there are a couple things I can point to that I think I would probably end up reverting in a refactor, most of my objection boils down to "this is a lot of change in code for a very small change in functionality".

**Comment by effigies** [comment]
And everything is Python 3. I have no intention to support Python 2, though it might still be compatible.

**Comment by adelavega** [comment]
If you what you suggested works, then that works for me for now.

Having said that, of the two options here (allowing `main` to be imported vs allowing `create_workflow` to be imported), I would prefer the latter because it would allow me to use it like a regular python function, rather than having to craft a list of strings as such:

     `main(['/bids/dir', '/output/dir', 'analysis_level', '--nthreads=1'])`

But maybe we can worry about that later and just go with the simple thing for now.

**Comment by effigies** [comment]
Oh, it looked like you were preferring `main` to other targets. So just skipping the shell, but otherwise treating it like a `subprocess.Popen` call rather than a function call. Which made sense to me, because I think `main` will be a more stable API than `create_workflow` for a little while.

So maybe let's go with this for right now. If we do a refactor now, I'm very likely to want to do another once the end goal is clearer in my head, and then we'd have to worry about me breaking your interface when I do go about that.

**Comment by adelavega** [comment]
Yeah that makes sense. I'll test that your solution works and push it soon.

**Comment by adelavega** [comment]
OK, tested it and it seems to work!

**Comment by effigies** [comment]
Ah, I feel like I've seen pathlib, but I haven't taken the trouble to learn it yet.

**Pull request merged**